numpy/core/tests/test_ufunc.py

```diff
@@ -2002,6 +2002,23 @@ def test_inplace_fancy_indexing(self):
         # Test multiple output ufuncs raise error, gh-5665
         assert_raises(ValueError, np.modf.at, np.arange(10), [1])
 
+        # Test ufuncs with non-trivial signature raise a TypeError
+        a = np.ones((2, 2, 2))
+        b = np.ones((1, 2, 2))
+        assert_raises(TypeError, np.matmul.at, a, [0], b)
+
+        a = np.array([[[1, 2], [3, 4]]])
+        assert_raises(TypeError, np.linalg._umath_linalg.det.at, a, [0])
```

Comment:
These tests look good.  But single test functions getting too long tends to be inconvenient.  Could you make a new method for these two?  Could just be something mentioning that gufuncs don't support `at` or so...